R/pkg/R/DataFrame.R

```diff
@@ -303,8 +303,28 @@ setMethod("colnames",
 #' @rdname columns
 #' @name colnames<-
 setMethod("colnames<-",
-          signature(x = "DataFrame", value = "character"),
```

Comment:
The reason I added this was because of this:

> colnames(iris) <- 1
Error in `colnames<-`(`*tmp*`, value = 1) : 
  'dimnames' applied to non-array

After I ran that, I saw this:
> showMethods("colnames<-")
Function: colnames<- (package SparkR)
x="ANY", value="ANY"
x="DataFrame", value="character"
x="DataFrame", value="numeric"
    (inherited from: x="ANY", value="ANY")

So looks like R automatically adds definitions of colnames<- if value is other than character.

This does not happen with coltypes<-, as it's not part of base package and doesn't have an (ANY,ANY) signature.

> coltypes(irisDF) <- 1
Error in (function (classes, fdef, mtable)  : 
  unable to find an inherited method for function ‘coltypes<-’ for signature ‘"DataFrame", "numeric"’